### ISSUE TITLE
Update axe-core

### ISSUE THREAD

**Issue by mcoker** [patternfly/patternfly-a11y#32]
We're currently on `"axe-core": "^3.5.5"` and latest is `4.3.2`. We're using aria attributes in our components that aren't supported in the current version of axe, that are supported in the latest version. As result, we've added `aria-allowed-attr` to the `ignoreRules`.

I ran the a11y test against core locally with the latest version of axe, and it looks like there are a bunch of new errors we will need to resolve. I imagine this will the the case with the react repo, too.

After talking with @jessiehuff, @tlabaj, and @nicolethoen, we'd like to go ahead and update the version of axe if we can. Our notes about this were something like:

* Turn off A11y tests so they don’t fail the builds
* Evaluate how many tests would fail at the new versions
* Open issues for each failing test and begin incrementally fixing them
* Instruct the team in how to run a11y tests locally so they can be responsible for not introducing new failing tests

I would also be in favor of updating the version, and leaving the tests running, but keeping them from failing the build if we can. Then we can also monitor the list of failures in CI, and make sure we aren't introducing new errors.

@redallen wdyt would be the best approach? Any other issues here that we haven't considered?

**Comment by redallen** [comment]
Have you considered how long these tests take to run locally? It's 15 minutes in Core and 14 minutes in React.

**Comment by mcoker** [comment]
> I would also be in favor of updating the version, and leaving the tests running, but keeping them from failing the build if we can. Then we can also monitor the list of failures in CI, and make sure we aren't introducing new errors.

@redallen agree. Can we do ^? This is what I thought we agreed on from our meeting, though I may be mistaken. If so, that seems sufficient. Then devs can run the tests locally if they want.

**Comment by redallen** [comment]
Yes we can add an `|| true` to the a11y tests when bumping `@patternfly/patternfly-a11y` in Core/React. Otherwise you'd have to fix all the a11y issues in the bump PR.

I was just stating I don't think we can realistically add the `|| true` AND "Instruct the team in how to run a11y tests locally so they can be responsible for not introducing new failing tests."

**Comment by jessiehuff** [comment]
Well for running the tests themselves, I think it would be sufficient to run the Axe browser extension on whatever page they've touched. It typically takes less than 5 seconds for me so I imagine that shouldn't be too hard of a check for devs. (I can show everyone how to use the extension in any meeting.) I agree that running all of the tests locally would be a real pain.

**Comment by redallen** [comment]
Even if you used the extension you'd have to:

1. Make sure that the extension uses the same version of axe-core AND same config as `@patternfly/patternfly-a11y`
2. Test the page(s) you're changing before your PR, record the errors, and then test after your PR and diff the errors

This seems rather error-prone...

**Comment by redallen** [comment]
Plan:

**Comment by jessiehuff** [comment]
Sure, that sounds good to me! So it sounds like CI would basically still run the tests, just not fail the build?

**Comment by redallen** [comment]
@jessiehuff Yes.

**Comment by jschuler** [comment]
axe-core has been updated to a newer version
